### PR TITLE
[parliamentlive.tv] Fix extractor

### LINKED ISSUE #1139: [Broken] ParliamentLive.TV has changed streaming platforms
## Checklist

- [X] I'm reporting a broken site support
- [X] I've verified that I'm running yt-dlp version **2021.09.25**
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [X] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
$ yt-dlp --verbose  https://www.parliamentlive.tv/Event/Index/34be5e69-5133-4bcd-ae29-dd96908a4457
[debug] Command-line config: ['--verbose', 'https://www.parliamentlive.tv/Event/Index/34be5e69-5133-4bcd-ae29-dd96908a4457']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] yt-dlp version 2021.09.25 (zip)
[debug] Python version 3.7.3 (CPython 64bit) - Darwin-20.6.0-x86_64-i386-64bit
[debug] exe versions: ffmpeg 3.1.1, ffprobe 3.1.1
[debug] Optional libraries: sqlite
[debug] Proxy map: {}
[debug] [parliamentlive.tv] Extracting URL: https://www.parliamentlive.tv/Event/Index/34be5e69-5133-4bcd-ae29-dd96908a4457
[parliamentlive.tv] 34be5e69-5133-4bcd-ae29-dd96908a4457: Downloading webpage
ERROR: [parliamentlive.tv] 34be5e69-5133-4bcd-ae29-dd96908a4457: Unable to extract kaltura widget config; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/Users/sams/bin/yt-dlp/yt_dlp/extractor/common.py", line 585, in extract
    ie_result = self._real_extract(url)
  File "/Users/sams/bin/yt-dlp/yt_dlp/extractor/parliamentliveuk.py", line 32, in _real_extract
    webpage, 'kaltura widget config'), video_id)
  File "/Users/sams/bin/yt-dlp/yt_dlp/extractor/common.py", line 1149, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
yt_dlp.utils.RegexNotFoundError: Unable to extract kaltura widget config; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/Users/sams/bin/yt-dlp/yt_dlp/extractor/common.py", line 585, in extract
    ie_result = self._real_extract(url)
  File "/Users/sams/bin/yt-dlp/yt_dlp/extractor/parliamentliveuk.py", line 32, in _real_extract
    webpage, 'kaltura widget config'), video_id)
  File "/Users/sams/bin/yt-dlp/yt_dlp/extractor/common.py", line 1149, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
yt_dlp.utils.RegexNotFoundError: Unable to extract kaltura widget config; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/sams/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1227, in wrapper
    return func(self, *args, **kwargs)
  File "/Users/sams/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1252, in __extract_info
    ie_result = ie.extract(url)
  File "/Users/sams/bin/yt-dlp/yt_dlp/extractor/common.py", line 602, in extract
    e.msg, video_id=video_id, ie=self.IE_NAME, tb=e.traceback, expected=e.expected, cause=e.cause)
yt_dlp.utils.ExtractorError: [parliamentlive.tv] 34be5e69-5133-4bcd-ae29-dd96908a4457: Unable to extract kaltura widget config; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.

```

## Description

The ParliamentLive.TV site (video from the UK Parliament) has recently changed streaming platforms, and the existing extractor is broken as all the streaming code has been changed. This extractor should likely be removed/archived:
https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/extractor/parliamentliveuk.py

Separately, the new platform is USP which seems to have several extractors built in, and I'm not sure which is easiest to use with the new site. Examples:
https://www.parliamentlive.tv/Event/Index/34be5e69-5133-4bcd-ae29-dd96908a4457
https://www.parliamentlive.tv/Event/Index/39877885-7c59-4a9a-9f03-8549a083f03c

### PR DESCRIPTION
**Pull request by u-spec-png**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information
Fix #1139.

### CODE REVIEW

File: yt_dlp/extractor/parliamentliveuk.py [hunk 1]
```diff
@@ -11,33 +19,60 @@ class ParliamentLiveUKIE(InfoExtractor):
...
         'only_matching': True,
     }]
+    _DEVICE_ID = str(uuid.uuid4())
+    _DATA = json.dumps({'device': {
+        'deviceId': _DEVICE_ID,
+        'width': 653,
+        'height': 368,
+        'type': 'WEB',
+        'name': ' Mozilla Firefox 91'},
+        'deviceId': _DEVICE_ID}).encode('utf-8')
```
Comment (pukkandan) [inline yt_dlp/extractor/parliamentliveuk.py:42]: generating uuid should go inside `_real_initialize`

File: yt_dlp/extractor/parliamentliveuk.py [hunk 2]
```diff
@@ -11,33 +19,60 @@ class ParliamentLiveUKIE(InfoExtractor):
...
+        video_info = self._download_json(f'https://www.parliamentlive.tv/Event/GetShareVideo/{video_id}', video_id)
+
+        thumbnail = video_info.get('thumbnailUrl')
+        event_title = video_info['event']['title']
+        timestamp = unified_timestamp(try_get(video_info, lambda x: x['event']['publishedStartTime']))
+        auth = 'Bearer ' + self._download_json(
+            'https://exposure.api.redbee.live/v2/customer/UKParliament/businessunit/ParliamentLive/auth/anonymous',
+            video_id, headers={'Origin': 'https://videoplayback.parliamentlive.tv',
+                               'Accept': 'application/json, text/plain, */*',
+                               'Content-Type': 'application/json;charset=utf-8'}, data=self._DATA)['sessionToken']
```
Comment (pukkandan) [inline yt_dlp/extractor/parliamentliveuk.py:55]: Are you sure the accept and content-type are needed? Generally, the defaults are enough

File: yt_dlp/extractor/parliamentliveuk.py [hunk 3]
```diff
@@ -11,33 +19,60 @@ class ParliamentLiveUKIE(InfoExtractor):
...
+
+        video_urls = self._download_json(
+            f'https://exposure.api.redbee.live/v2/customer/UKParliament/businessunit/ParliamentLive/entitlement/{video_id}/play',
+            video_id, headers={'Authorization': auth, 'Accept': 'application/json, text/plain, */*'})['formats']
```
Comment (pukkandan) [inline yt_dlp/extractor/parliamentliveuk.py:59]: same

File: yt_dlp/extractor/parliamentliveuk.py [hunk 4]
```diff
@@ -11,33 +19,60 @@ class ParliamentLiveUKIE(InfoExtractor):
...
-        kaltura_url = 'kaltura:%s:%s' % (
-            widget_config['wid'][1:], widget_config['entry_id'])
-        event_title = self._download_json(
-            'http://parliamentlive.tv/Event/GetShareVideo/' + video_id, video_id)['event']['title']
+        video_info = self._download_json(f'https://www.parliamentlive.tv/Event/GetShareVideo/{video_id}', video_id)
+
+        thumbnail = video_info.get('thumbnailUrl')
+        event_title = video_info['event']['title']
+        timestamp = unified_timestamp(try_get(video_info, lambda x: x['event']['publishedStartTime']))
+        auth = 'Bearer ' + self._download_json(
```
Comment (pukkandan) [inline yt_dlp/extractor/parliamentliveuk.py:51]: move thumbnails, title and timestamp near the return or maybe even directly inside the return

File: yt_dlp/extractor/parliamentliveuk.py [hunk 5]
```diff
@@ -11,33 +19,60 @@ class ParliamentLiveUKIE(InfoExtractor):
...
+            video_id, headers={'Authorization': auth, 'Accept': 'application/json, text/plain, */*'})['formats']
+
+        formats = []
+        for format in video_urls:
+            if format.get('format') == 'DASH' or format.get('format') == 'SMOOTHSTREAMING':
+                formats.extend(self._extract_mpd_formats(
+                    format.get('mediaLocator'), video_id, mpd_id='dash', fatal=False))
+            elif format.get('format') == 'HLS':
+                formats.extend(self._extract_m3u8_formats(
+                    format.get('mediaLocator'), video_id, m3u8_id='hls'))
```
Comment (pukkandan) [inline yt_dlp/extractor/parliamentliveuk.py:68]:
```suggestion
            if not format.get('mediaLocator'):
                continue
            if format.get('format') in ('DASH', 'SMOOTHSTREAMING'):
                formats.extend(self._extract_mpd_formats(
                    format['mediaLocator'], video_id, mpd_id='dash', fatal=False))
            elif format.get('format') == 'HLS':
                formats.extend(self._extract_m3u8_formats(
                    format['mediaLocator'], video_id, m3u8_id='hls', fatal=False))
```

btw, are you sure SMOOTHSTREAMING is not ism but is mpd? that would be weird

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/parliamentliveuk.py (hunk 2 above)
Comment (u-spec-png) [reply yt_dlp/extractor/parliamentliveuk.py:55]: Yes, they are all needed.

File: yt_dlp/extractor/parliamentliveuk.py [hunk 6]
```diff
@@ -25,19 +35,46 @@ class ParliamentLiveUKIE(InfoExtractor):
...
+            'height': 368,
+            'type': 'WEB',
+            'name': ' Mozilla Firefox 91'},
+            'deviceId': _DEVICE_ID}).encode('utf-8')
+
+        auth = 'Bearer ' + self._download_json(
+            'https://exposure.api.redbee.live/v2/customer/UKParliament/businessunit/ParliamentLive/auth/anonymous',
+            video_id, headers={'Origin': 'https://videoplayback.parliamentlive.tv',
+                               'Accept': 'application/json, text/plain, */*',
+                               'Content-Type': 'application/json;charset=utf-8'}, data=_DATA)['sessionToken']
```
Comment (pukkandan) [inline yt_dlp/extractor/parliamentliveuk.py:52]:
```suggestion
        auth = 'Bearer ' + self._download_json(
            'https://exposure.api.redbee.live/v2/customer/UKParliament/businessunit/ParliamentLive/auth/anonymous',
            video_id, headers={
                'Origin': 'https://videoplayback.parliamentlive.tv',
                'Accept': 'application/json, text/plain, */*',
                'Content-Type': 'application/json;charset=utf-8'
            }, data=json.dumps({
                'deviceId': _DEVICE_ID,
                'device': {
                    'deviceId': _DEVICE_ID,
                    'width': 653,
                    'height': 368,
                    'type': 'WEB',
                    'name': ' Mozilla Firefox 91'
                }
            }).encode('utf-8'))['sessionToken']
```

File: yt_dlp/extractor/parliamentliveuk.py [hunk 7]
```diff
@@ -25,19 +35,49 @@ class ParliamentLiveUKIE(InfoExtractor):
...
+                'Accept': 'application/json, text/plain, */*',
+                'Content-Type': 'application/json;charset=utf-8'
+            }, data=json.dumps({
+                'deviceId': _DEVICE_ID,
+                'device': {
+                    'deviceId': _DEVICE_ID,
+                    'width': 653,
+                    'height': 368,
+                    'type': 'WEB',
+                    'name': ' Mozilla Firefox 91'
```
Comment (pukkandan) [inline yt_dlp/extractor/parliamentliveuk.py:53]: Is the space before Mozilla intentional?

Comment (u-spec-png) [reply yt_dlp/extractor/parliamentliveuk.py:53]: Yes, but it's not necessary I just copied POST request data and it was like that, I just didn't test it to see if it is necessary.

**Pull request merged**